decode.go

```diff
@@ -299,6 +299,28 @@ func (p *parser) mapping() *Node {
 // ----------------------------------------------------------------------------
 // Decoder, unmarshals a node into a provided value.
 
+// DecodeConfig represents configuration for decoding.
+// Config could be initialized in "builder" style for convenience:
+// NewDecodeConfig().KnownFields(true).UniqueFields(true)
+type DecodeConfig struct {
```

Comment:
Options might be better than Config, as the former is a real word.

Reply:
Renamed to DecodeOptions